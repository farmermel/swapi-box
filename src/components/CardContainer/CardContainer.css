.card-container {
  display: flex;
  flex-wrap: wrap; }

.loading-gif {
  border: 1px solid cornflowerblue;
  margin: auto;
  width: 500px; }
