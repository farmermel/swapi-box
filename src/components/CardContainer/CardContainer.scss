.card-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  // display: flex;
  // // overflow-x: scroll;
  // flex-wrap: wrap;
}

.loading-gif {
  border: 1px solid cornflowerblue;
  margin: auto;
  width: 500px;
  grid-column-start: 2;
  grid-column-end: 3;
}

@media screen and (max-width: 1140px) {
  .card-container {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media screen and (max-width: 850px) {
  .card-container {
    grid-template-columns: 1fr 1fr;
  }
}

@media screen and (max-width: 500px) {
  .card-container {
    grid-template-columns: 1fr;
  }
}