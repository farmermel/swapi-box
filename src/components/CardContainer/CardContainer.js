import React from 'react';
import PeopleCard from '../PeopleCard/PeopleCard';
import PlanetCard from '../PlanetCard/PlanetCard';
import VehicleCard from '../VehicleCard/VehicleCard';
import FavoritesCard from '../FavoritesCard/FavoritesCard';
import loading from '../../assets/loading.gif';
import './CardContainer.css';

let cards;

const cardTypes = {
  PeopleCard: PeopleCard,
  PlanetCard: PlanetCard,
  VehicleCard: VehicleCard,
  FavoritesCard: FavoritesCard
}

const createCards = (data, cardType) => {
  let Type = cardTypes[cardType];
  cards = data.map( card => {
    return <Type data={card} />
  })
} 

const CardContainer = ({ cardData, getData, cardType }) => {
  createCards(cardData, cardType)
  return (
    <div className='card-container'>
      {!cardData.length && <img src={loading} 
                                alt='page is loading'
                                className='loading-gif' /> }
      {cards}
    </div>
  )
}

export default CardContainer;