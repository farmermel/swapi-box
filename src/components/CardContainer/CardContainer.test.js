import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';
import mockData from '../../mock-data';

describe('CardContainer', () => {
  it('matches the snapshot', () => {
    const cardData = mockData.cleaned.planet;
    let wrapper = shallow(<CardContainer cardData={cardData}
                                toggleFav={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })
})