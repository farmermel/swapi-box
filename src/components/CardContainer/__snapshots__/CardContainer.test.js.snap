// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardData={
        Array [
            Object {
              "favorite": false,
              "id": 9823742,
              "info": Object {
                "climate": "hot",
                "name": "NABOO",
                "population": "4",
                "residents": "no residents",
                "terrain": "sandy",
              },
              "type": "planetData",
            },
          ]
    }
    toggleFav={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <Card
            data={
                        Object {
                                    "favorite": false,
                                    "id": 9823742,
                                    "info": Object {
                                      "climate": "hot",
                                      "name": "NABOO",
                                      "population": "4",
                                      "residents": "no residents",
                                      "terrain": "sandy",
                                    },
                                    "type": "planetData",
                                  }
            }
            toggleFav={[Function]}
/>,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "9823742",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "favorite": false,
            "id": 9823742,
            "info": Object {
              "climate": "hot",
              "name": "NABOO",
              "population": "4",
              "residents": "no residents",
              "terrain": "sandy",
            },
            "type": "planetData",
          },
          "toggleFav": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <Card
              data={
                            Object {
                                          "favorite": false,
                                          "id": 9823742,
                                          "info": Object {
                                            "climate": "hot",
                                            "name": "NABOO",
                                            "population": "4",
                                            "residents": "no residents",
                                            "terrain": "sandy",
                                          },
                                          "type": "planetData",
                                        }
              }
              toggleFav={[Function]}
/>,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "9823742",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "favorite": false,
              "id": 9823742,
              "info": Object {
                "climate": "hot",
                "name": "NABOO",
                "population": "4",
                "residents": "no residents",
                "terrain": "sandy",
              },
              "type": "planetData",
            },
            "toggleFav": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
